frontend/src/store.ts

```diff
@@ -5130,7 +5190,156 @@ export function createStore(web3: Web3) {
         if(!BurningManager || !state.defaultAccount) return;
 
         return await BurningManager.methods.vars(2).call(defaultCallOptions(state));
+      },
+
+      async fetchSpecialWeaponEvents({ state, dispatch, commit }) {
+        const { SpecialWeaponsManager } = state.contracts();
+        if(!SpecialWeaponsManager || !state.defaultAccount) return;
+
+        const activeSpecialWeaponEventsIds = await SpecialWeaponsManager.methods.getActiveEventsIds().call(defaultCallOptions(state));
+        commit('updateActiveSpecialWeaponEventsIds', activeSpecialWeaponEventsIds);
+
+        const eventCount = await SpecialWeaponsManager.methods.eventCount().call(defaultCallOptions(state));
+        const inactiveSpecialWeaponEventsIds = Array.from({length: +eventCount}, (_, i) =>
+          (i + 1).toString()).filter(id => !activeSpecialWeaponEventsIds.includes(id));
+        commit('updateInactiveSpecialWeaponEventsIds', inactiveSpecialWeaponEventsIds);
+
+        await dispatch('fetchSpecialWeaponEventsInfo', activeSpecialWeaponEventsIds.concat(inactiveSpecialWeaponEventsIds));
+        await dispatch('fetchShardsSupply');
+      },
+
+      async fetchSpecialWeaponEventsInfo({ state, commit }, eventsIds) {
+        const { SpecialWeaponsManager } = state.contracts();
+        if(!SpecialWeaponsManager || !state.defaultAccount) return;
+
+        eventsIds.forEach(async (eventId: string) => {
+          const eventInfoRaw = await SpecialWeaponsManager.methods.getEventInfo(eventId).call(defaultCallOptions(state));
+          const ordered = +await SpecialWeaponsManager.methods.userOrderOptionForEvent(state.defaultAccount!, eventId).call(defaultCallOptions(state)) > 0;
+          const forged = await SpecialWeaponsManager.methods.userForgedAtEvent(state.defaultAccount!, eventId).call(defaultCallOptions(state));
+          const eventInfo = {
+            name: eventInfoRaw[0],
+            weaponElement: eventInfoRaw[1],
+            endTime: eventInfoRaw[2],
+            supply: eventInfoRaw[3],
+            orderedCount: eventInfoRaw[4],
+            ordered,
+            forged
+          };
+
+          commit('updateSpecialWeaponEventsInfo', { eventId, eventInfo });
+        });
+      },
+
+      async fetchForgeCosts({ state }) {
+        const { SpecialWeaponsManager } = state.contracts();
+        if(!SpecialWeaponsManager || !state.defaultAccount) return;
+
+        const shardsCostLow = await SpecialWeaponsManager.methods.vars(1).call(defaultCallOptions(state));
```

Comment:
Doesn't matter that much, because the vars won't probably change the values, but you can use constants here (from the contract)

Reply:
I wanted to avoid extra loading time.